no matching function for call to 'MusicSequenceFileLoadData'

Hello,

The latest raceintospace commit as of right now (d6b4649b16c3df4128942e3fe778193e47a7b7b4) doesn't build on macOS 10.13.6. The error (after applying the fix from #306) is:

```
/path/to/src/game/platform_macosx/music_osx.cpp:172:14: error: no matching function for call to 'MusicSequenceFileLoadData'
        rv = MusicSequenceFileLoadData(music_files[track].sequence, CFDataCreate(NULL, (UInt8*)midi_data, fileLength), 0, 0);
             ^~~~~~~~~~~~~~~~~~~~~~~~~
/System/Library/Frameworks/AudioToolbox.framework/Headers/MusicPlayer.h:816:1: note: candidate function not viable: no known conversion from 'int' to 'MusicSequenceFileTypeID' for 3rd argument
MusicSequenceFileLoadData (MusicSequence                        inSequence,
^
1 error generated.
```

--- a/src/game/platform_macosx/music_osx.cpp
+++ b/src/game/platform_macosx/music_osx.cpp
@@ -35,6 +35,11 @@ extern "C" {
 #include "music.h"
 #include "logging.h"
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1090
+#define kMusicSequenceFile_AnyType 0
+#define kMusicSequenceLoadSMF_PreserveTracks 0
+#endif
+
 LOG_DEFAULT_CATEGORY(music);
 
 // A map of music_tracks to filenames
@@ -167,9 +172,9 @@ void music_load(enum music_track track)
 	if (rv == 0) {
 		// Try to load the sequence out of this buffer
 #if MAC_OS_X_VERSION_MIN_REQUIRED <= MAC_OS_X_VERSION_10_4 /* this is deprecated, but works back to 10.3 */        
-		rv = MusicSequenceLoadSMFDataWithFlags(music_files[track].sequence, CFDataCreate(NULL, (UInt8*)midi_data, fileLength), 0);
+		rv = MusicSequenceLoadSMFDataWithFlags(music_files[track].sequence, CFDataCreate(NULL, (UInt8*)midi_data, fileLength), kMusicSequenceLoadSMF_PreserveTracks);
 #else  /* not deprecated, but requires 10.5 or later */
-        rv = MusicSequenceFileLoadData(music_files[track].sequence, CFDataCreate(NULL, (UInt8*)midi_data, fileLength), 0, 0);
+		rv = MusicSequenceFileLoadData(music_files[track].sequence, CFDataCreate(NULL, (UInt8*)midi_data, fileLength), kMusicSequenceFile_AnyType, kMusicSequenceLoadSMF_PreserveTracks);
 #endif
 	}
 		

